src/Cli/dotnet/ReleasePropertyProjectLocator.cs

```diff
@@ -280,10 +280,26 @@ private bool IsValidSlnFilePath(string path)
 
             string[] globalPropEnumerable = _parseResult.GetValue(CommonOptions.PropertiesOption);
 
-            foreach (var keyEqVal in globalPropEnumerable)
+            foreach (var keyEqValString in globalPropEnumerable)
             {
-                string[] keyValuePair = keyEqVal.Split("=", 2);
-                globalProperties[keyValuePair[0]] = keyValuePair[1];
+                foreach (var keyEqVal in keyEqValString.Split(";"))
+                {
+                    string[] keyValuePair = keyEqVal.Split("=", 2);
+                    if (keyValuePair.Count() == 0)
+                    {
+                        // The user provided an empty value after ;. e.g. -p:Foo=Bar;
+                        continue;
+                    }
+                    else if (keyValuePair.Count() == 1)
+                    {
+                        // The user provided a property such as -p:Foo without giving a value. Undefine the property.
+                        // Note that MSBuild will fail if given -p:Foo; but we can ignore this situation because of this and let MsBuild fail, or, assuming it comes to parity with ';', behave the same as if ; was not present.
+                        // The same can be said for -p:Foo=bar;-p:Bar=Foo, which is also invalid syntax to MSBuild.
+                        globalProperties[keyValuePair[0]] = "";
+                        continue;
+                    }
+                    globalProperties[keyValuePair[0]] = keyValuePair[1];
```

Comment:
I'd suggest just using if/else logic here and skipping the `continue` statements:

```suggestion
                    if (keyValuePair.Count() == 0)
                    {
                        // The user provided an empty value after ;. e.g. -p:Foo=Bar;
                    }
                    else if (keyValuePair.Count() == 1)
                    {
                        // The user provided a property such as -p:Foo without giving a value. Undefine the property.
                        // Note that MSBuild will fail if given -p:Foo; but we can ignore this situation because of this and let MsBuild fail, or, assuming it comes to parity with ';', behave the same as if ; was not present.
                        // The same can be said for -p:Foo=bar;-p:Bar=Foo, which is also invalid syntax to MSBuild.
                        globalProperties[keyValuePair[0]] = "";
                    }
                    else
                    {
                        globalProperties[keyValuePair[0]] = keyValuePair[1];
                    }
```

Reply:
This is a 10/10 suggestion but the code has changed so not doing this now